db/db_impl/db_impl.h

```diff
@@ -1057,12 +1057,14 @@ class DBImpl : public DB {
   // unscheduled (queued, but not scheduled yet). If the db is shutting down,
   // Status::ShutdownInProgress will be returned. If PauseBackgroundWork() was
   // called prior to this, this may potentially wait for unscheduled jobs
-  // indefinitely. abort_on_pause can be set to true to abort, and
+  // indefinitely. abort_on_pause in the option can be set to true to abort, and
   // Status::Aborted will be returned immediately. This may also never return if
   // there's sufficient ongoing writes that keeps flush and compaction going
   // without stopping. The user would have to cease all the writes to DB to make
   // this eventually return in a stable state.
-  Status WaitForCompact(bool abort_on_pause = false);
+  Status WaitForCompact() { return WaitForCompact(WaitForCompactOptions()); }
```

Comment:
Wondering if there's a better way to do this.

Reply:
In this draft we have four variants of WaitForCompact in DBImpl (two with TEST_, two without). In the public API, we commonly expect an explicit WhateverOptions object to help prompt the users to think about the exact behavior they want, so it makes sense to expect WaitForCompactOptions always. Internally, it can get verbose in unit testing. Perhaps just keep TEST_WaitForCompact() with no parameters and the public API WaitForCompact.